### PR TITLE
feat: changes blockchain connection workflow

### PR DESCRIPTION
**Pull request by tjwelde**
## fixes KILTProtocol/ticket#1077
This PR removes the host parameter for `connect` and `getCached` and renames functions to clarify, what they actually do.
It optimises the disconnect function, so it does not start a new connection, when there was non before and adds a setConnection and getConnection function for advanced use-cases.
It also adds jsdoc for the `BlockchainApiConnection` functions.

NEEDS https://github.com/KILTprotocol/sdk-js/pull/344 TO BE MERGED, THAN THIS PR SHOULD TARGET `develop`.

AFTER MERGE WE CAN REMOVE https://github.com/KILTprotocol/sdk-js/compare/disconnect BRANCH.

## How to test:
- Run tests

## Checklist:

- [x] I have verified that the code works
- [x] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [x] I have [left the code in a better state](https://deviq.com/boy-scout-rule/)
- [x] I have documented the changes (where applicable)

### CODE REVIEW

File: packages/chain-helpers/src/blockchainApiConnection/BlockchainApiConnection.ts [hunk 1]
```diff
@@ -59,15 +65,66 @@ export async function buildConnection(
...
+export function getConnection(): Blockchain | null {
+  return instance
+}
+
+/**
+ * Gets the cached blockchain connection, or builds a new one, if non-existant.
+ *
+ * @returns The cached or newly built blockchain connection instance.
+ */
+export async function getConnectionOrConnect(): Promise<Blockchain> {
```
[the remark this answers is not in the document]
Comment (tjwelde) [reply packages/chain-helpers/src/blockchainApiConnection/BlockchainApiConnection.ts:94]: yeah, I am also torn.

File: packages/chain-helpers/src/blockchainApiConnection/BlockchainApiConnection.ts [hunk 2]
```diff
@@ -59,15 +65,66 @@ export async function buildConnection(
...
   if (!instance) {
-    instance = buildConnection(host)
+    instance = await buildConnection()
```
[the remark this answers is not in the document]
Comment (tjwelde) [reply packages/chain-helpers/src/blockchainApiConnection/BlockchainApiConnection.ts:96]: hmm, yeah. Maybe we should just go back to storing it as a promise? We use async function `getConnectionOrConnect ` either way everywhere so for most of our use cases it is not relevant to store the instance directly...

[the remark this answers is not in the document]
Comment (rflechtner) [reply packages/chain-helpers/src/blockchainApiConnection/BlockchainApiConnection.ts:96]: we probably should, the only other solution would be storing it as a promise AND resolved result - but that seems overkill

[the remark this answers is not in the document]
Comment (tjwelde) [reply packages/chain-helpers/src/blockchainApiConnection/BlockchainApiConnection.ts:101]: I reverted it

**Review by rflechtner — APPROVED** [review]
LGTM !

**Pull request merged**